src/core/api/clock.ts

```diff
@@ -292,11 +295,15 @@ function getStalledStatus(
   if (shouldUnstall === true) {
     return null;
   } else if (shouldStall === true || prevStalled !== null) {
-    let reason : "seeking" | "not-ready" | "buffering";
+    let reason : "seeking" | "not-ready" | "buffering" | "internal-seek";
     if (currentState === "seeking" ||
         currentTimings.seeking ||
         prevStalled !== null && prevStalled.reason === "seeking") {
-      reason = "seeking";
+          if (internalSeek) {
```

Comment:
We try to always check boolean as less mistakes can slip up.
For example, your condition doesn't work if we seeked at the position `0`.

Reply:
Yes u are right, but I reverted back to a `boolean`.